07week/towersOfHanoi/script.js

```diff
@@ -3,20 +3,104 @@
 class TowersOfHanoi extends React.Component {
   constructor(props) {
     super(props);
+    this.state = {
+      stack1: [100,75,50,25],
+      stack2: [],
+      stack3: [],
+      hand: null,
+      status: 'active'
+    }
+    this.moveBlock = this.moveBlock.bind(this);
+    this.drawBlocks = this.drawBlocks.bind(this);
+    this.removeBlock = this.removeBlock.bind(this);
+    this.placeBlock = this.placeBlock.bind(this);
+    this.checkWin = this.checkWin.bind(this);
   }
 
+checkWin() {
+  if (this.state.stack3.length == 4) {
+    const game = {
+      status: 'over'
+    }
+    this.setState(game);
+    return true;
+  } else {
+    return false;
+  }
+}
+
+drawBlocks(thisStack) {
+  console.log("drawBlocks test");
+  let list = [];
+  for (var b=0; b<thisStack.length; b++) {
+    let blockNum =  (thisStack[b]);
+    var div = <div data-block={blockNum}></div>
+    list.push(div);
+  }
+  return list
+}
+
+removeBlock(stack) {
+  let take = stack.pop();
+  const taken = {
+    hand: take
+  }
+  this.setState(taken);
+}
+
+placeBlock(stack) {
+  let drop = this.state.hand;
+  stack.push(drop);
+  console.log(stack);
+  const dropped = {
+    hand: null
+  }
+  this.setState(dropped);
+}
+
+moveBlock(event){
+  if (this.state.status === 'active'){
```

Comment:
you dont let me make an invlaid move, but a message telling me that would have been nice.

Nice job